### ISSUE TITLE
Is it possible to get tachyons way of inline styling with styled-system?

### ISSUE THREAD

**Issue by nikitavoloboev** [styled-system/styled-system#672]
I would love if I could add inline styles like what [tachyons](https://github.com/tachyons-css/tachyons) allowed you to do:

`<div className="flex flex-row items-center mb2 />`

Instead of doing:

`<div style={{display: 'flex', flexDirection: 'flex-row', ...}} />`

Is it possible to achieve with either styled-system or styled-components?

Or I am forced to load in Tachyons in order to get that ease of applying styles without all the boilerplate of styled-system?

**Comment by jxnblk** [comment]
Yes, have you read the Getting Started page? https://styled-system.com/getting-started/

This generates CSS with libraries like Styled Components, but you could also use it for inline  styles. Tachyons is also CSS (not inline styles) fwiw

**Comment by nikitavoloboev** [comment]
> have you read the Getting Started page?

I have but in order to add these I have to basically make everything into a component.

For example:


```
const Search: React.FC<Props> = props => {

  return (
    <div
// Here I wish I could just use Tachyons style annotating over style (like className)
      style={{
        display: "flexbox",
        flexDirection: "column",
        border: "1px solid black",
        padding: "4px 6px",
        width: "300px"
      }}
    >
      <input
        style={{ backgroundColor: "#F2F2F2", width: "300px", height: "30px" }}
        type="search"
        autoFocus
        placeholder="Search..."
      />
      <SearchOption
        name=""
        description=""
      />
    </div>
  );
};

export default styled(Search)`

`;
```

With styled-system I can't just do that. 

I will have to make a new styled-component and with styled-system add stuff like `${space}` or make a new div rather with `div.defaultProps` and write my styles there. That's a lot of separation and boilerplate.

Unless I misunderstand something.

**Comment by jxnblk** [comment]
Styled System is a lower-level library that is meant for building components like this, but you might be interested in either [Rebass](https://rebassjs.org), which is a set of components with the style props already built, or [Theme UI](https://theme-ui.com), which allows you to do what you're describing above with the special [`sx` prop](https://theme-ui.com/sx-prop/)

**Comment by nikitavoloboev** [comment]
> be interested in either [Rebass](https://rebassjs.org)

So instead of using typical components like `div`, `input` box, I pull things from `Rebass`?

I thought `Rebass` doesn't cover all of HTML elements. So you will still need to use raw html elements and you will still have the need to inline style them.

**Comment by nikitavoloboev** [comment]
I am curious if during development, you never use `style={{}}` anymore to style things?

**Comment by jxnblk** [comment]
> I thought Rebass doesn't cover all of HTML elements. 

You can use the `as` prop to change the underlying element, e.g. `<Box as='input' />`, but it really sounds like you'd be more interested in Theme UI, based on your code examples.

> I am curious if during development, you never use style={{}} anymore to style things?

Inline styles are useful for dynamic style changes, but don't cover everything possible with CSS, such as pseudoclasses, media queries, etc., so it's not a full replacement

**Comment by nikitavoloboev** [comment]
> you'd be more interested in Theme UI

Reading through its readme, I see it uses Emotion under the hood which is practically the same thing as styled-components CSS in JS library. 

I am currently using styled-components, I assume by using Theme UI, I will have to stop using it as it doesn't make sense to have two conflicting things being loaded. 

Also not sure how it impacts the performance of generated CSS and thus the speed of the app.

**Comment by jxnblk** [comment]
Yeah with Theme UI, it uses Emotion, so you probably wouldn't want to mix that library with Styled Components. You can also use this utility with the `css` prop for a similar API with Styled Components https://styled-system.com/css/
